test/src/test/java/org/springframework/security/test/context/support/ReactorContextTestExecutionListenerTests.java

```diff
@@ -174,6 +174,11 @@ public void afterTestMethodWhenSetupThenReactorContextNull() throws Exception {
 		assertThat(Mono.subscriberContext().block().isEmpty()).isTrue();
 	}
 
+	@Test
+	public void afterTestMethodWhenDifferentHookIsRegistered() throws Exception {
+		this.listener.afterTestMethod(this.testContext);
+	}
```

Comment:
Thanks @rwinch. I will look at this and make corrections. Would I need to create another PR when I'm done?

Reply:
Hello @rwinch. 

I am trying to register a `Hooks.onLastOperator.` then I realised I do not have access to `DelegateTestExecutionListener` in the `ReactorContextTestExecutionListener` class since it is private. The only way I felt I could register it is by calling the overriden `beforeTestMethod` but it's also declared in a private static class.

I'm a bit confused as to how things are expected to go. 

This is what I have at the moment:
```
	@Test
	public void afterTestMethodWhenDifferentHookIsRegistered() throws Exception {

		SecurityContext securityContext = TestSecurityContextHolder.getContext();

		String CONTEXT_OPERATOR_KEY = SecurityContext.class.getName();

		this.listener.beforeTestMethod(this.testContext);

		//Register Hook Here
		
		this.listener.afterTestMethod(this.testContext);

		Hooks.resetOnLastOperator(CONTEXT_OPERATOR_KEY);
	}
```

Also, in order to validate if the hook is still present (i.e. after the hook has been registered), I'm thinking to use an `assertThat().isEqualTo()` but I don't know what value to validate with.

I hate to bother you but would really appreciate your input. Thanks a lot.

Reply:
@daejii I think what we want here is to register a custom hook and confirm that it is still invoked.

For example, something like this:

```java
public void afterTestMethodWhenDifferentHookIsRegistered() throws Exception {
    Function custom = mock(Function.class);
    when(custom.apply(any(Object.class))).thenAnswer(invocation -> invocation.getArgument(0));

    Hooks.onLastOperator("custom", custom);
    this.listener.afterTestMethod(this.testContext);
    
    Mono.subscriberContext.block();
    verify(custom).apply(any(Object.class));
}
```

@rwinch may have a slicker way, but the idea here is that want to make sure that ReactorContextTestExecutionListener is no longer overriding a custom hook, so we need a custom hook in our test.

Also, `cleanup` already called `Hooks.resetOnLastOperator()`. So I suppose that calling `Hooks.resetOnLastOperator("custom")` is not necessary.

Reply:
Hello @jzheaux Thanks a lot for your response. 
By "custom", did you mean I could use any random string value as opposed to using `SecurityContext.class.getName()`.

Reply:
Also, by `Object obj = new Object();` Did you mean the Object class itself or an object like `SecurityContext`? Thanks @jzheaux

Reply:
@daejii yes, any custom-named hook will do. Do not use `SecurityContext.class.getName()` since this test is trying to add a hook *in addition to* the one that `ReactorContextTestExecutionListener` adds.

I actually did mean `Object obj = new Object();`. See at the end that we confirm that `obj` was what was in the subscriber context, confirming that our custom hook was invoked (as opposed to clobbered, like would have happened before your fix).

Reply:
@jzheaux Okay. I get it now. Thanks a lot for your input. I really appreciate the opportunity. Means a lot to me. I will submit a PR soon.

Reply:
Hello @jzheaux @rwinch. I have created a new PR. #6174. Kindly review. Thanks.